### ISSUE TITLE
Pie chart issue - gets cut off at smaller screens OR is too big for container at bigger screens

### ISSUE THREAD

**Issue by xenia1** [apexcharts/vue-apexcharts#464]
Hi, 

Tl;dr - pie chart resizing within fixed height box works fine when the legend shows but not when legend is hidden

--
### Details: 

I have an issue with pie charts sizing within a box that has a fixed height, e.g. 300px (variable width)

When the legend shows, the pie chart resizing as the screen changes size works fine.

_Without_ the legend (I added a button/method to toggle its visibility), one of two things happens: 

1. Give pie chart height of 100%, ie.: 

```
      <apexcharts
        type="pie"
        :height="'100%'"
        :options="options"
        :series="series"
      ></apexcharts>
```

**Result** - as the screen / box gets narrower, the sides of the pie chart get cropped.

_Visual_:

<img width="259" alt="Screen Shot 2023-01-23 at 12 11 10 PM" src="https://user-images.githubusercontent.com/14450967/214043194-d0f20f57-d260-4f57-a050-607a230c0fd2.png">

2. Give pie chart empty height, i.e. : 
```
      <apexcharts
        type="pie"
        :height="''" // OR omit this line completely
        :options="options"
        :series="series"
      ></apexcharts>
```

**Result** - When the box is narrow, the box resizes correctly. BUT when it's wider, the pie chart is way too tall for the box and spills outside.

_Visual_:

NARROW SCREEN (desired behaviour):
<img width="229" alt="Screen Shot 2023-01-23 at 1 05 36 PM" src="https://user-images.githubusercontent.com/14450967/214046968-ea43e035-c5e4-4a19-b66b-7ba90888cd60.png">

WIDE SCREEN (not desired behaviour):
<img width="624" alt="Screen Shot 2023-01-23 at 12 11 29 PM" src="https://user-images.githubusercontent.com/14450967/214043274-f35eda93-80cc-4555-96bb-c9bf4121e0f7.png">

I tried to replicate the issue, however it's not a perfect replication for the following reason:

With empty height, in my code, when I toggle the legend on/off the chart will resize to be _correct with the legend_ but _overflow without the legend_. In the fiddle, the FIRST time I hide the legend, the chart becomes too big. But then showing the legend again doesn't make it reset properly. I cannot figure out why the behaviour is not the same. 

If the height is 100%, the behaviour is the same in my code and in the fiddle - then the sides get cropped if the box becomes narrow.

**Fiddle**:
https://codesandbox.io/s/vue-basic-example-forked-xnj1h4?file=/src/components/Chart.component.vue

**Conclusion:**
Does anyone have any advice on what kind of height value I can give the chart/what kind of code structure I can have so that when the legend is hidden, the chart sized correctly and adjusts as the screen width changes?

In other words, is there any way that when the chart (without a legend) is in a fixed height box, it can fit into that box and resize correctly according to the width of the box? 

Can the resizing can work the same whether the legend is visible or not? 

Thanks!  And let me know if you require any more details.
